### PR TITLE
Bot movement.

### LINKED ISSUE #20: Bots' improvemed movement.
Made it while ago for 1.2 version. I can make it there but it'll take some time for me to disassemble required functions. What do you think? Is it a good idea or not?
[Youtube](https://www.youtube.com/watch?v=XPdf4L6p_Cc)

### PR DESCRIPTION
**Pull request by T-Maxxx**
resolved #20

### CODE REVIEW

File: src/q_math.c [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #include "q_math.h"
 #include <math.h>
 
+void (__cdecl *Math_VectorToAngles)(vec3_t vector, vec3_t angles) = (void(__cdecl*)(vec3_t, vec3_t))0x081921A2;
```
Comment (IceNinjaman) [inline src/q_math.c]: Please don't reference simple functions like this one especially when you can get the sourcecode of it.

`
void vectoangles( const vec3_t value1, vec3_t angles ) {
	float	forward;
	float	yaw, pitch;
	
	if ( value1[1] == 0 && value1[0] == 0 ) {
		yaw = 0;
		if ( value1[2] > 0 ) {
			pitch = 90;
		}
		else {
			pitch = 270;
		}
	}
	else {
		if ( value1[0] ) {
			yaw = ( atan2 ( value1[1], value1[0] ) * 180 / M_PI );
		}
		else if ( value1[1] > 0 ) {
			yaw = 90;
		}
		else {
			yaw = 270;
		}
		if ( yaw < 0 ) {
			yaw += 360;
		}

		forward = sqrt ( value1[0]*value1[0] + value1[1]*value1[1] );
		pitch = ( atan2(value1[2], forward) * 180 / M_PI );
		if ( pitch < 0 ) {
			pitch += 360;
		}
	}

	angles[PITCH] = -pitch;
	angles[YAW] = yaw;
	angles[ROLL] = 0;
}`

**Comment by T-Maxxx** [comment]
Done.

**Comment by PureTryOut** [comment]
I'm guessing these last commits still have something to do with this PR? If not, you should look into working in branches for each feature/fix next time.

**Comment by T-Maxxx** [comment]
There's few unnamed function pointers and one huge function left to be reversed. Its still works, but I have no time in few days to work on it.
And if somebody have "DObj" structure reversed, I'll be grateful.

**Comment by IceNinjaman** [comment]
I think to have DObj structure quite accurate. I can maybe add it tomorrow.
Which large function you need? If you have address of iw3mp.exe I can see.

**Comment by T-Maxxx** [comment]
0x081AE114. Its not hard to reverse, but I have no time to do that. That's what I know about DObj at this moment. Xrefs for `GetDObjForEntity` (0x08125E32) can help you.

**Comment by IceNinjaman** [comment]
Sent email at tm*********5@gmail.com with a set of header files. I hope you received it xD

**Comment by T-Maxxx** [comment]
I believe thats it. Deeper RE seems like waste of time.

File: plugins/legacybanlist/q_shared.h [hunk 2]
```diff
@@ -456,7 +456,7 @@ void MakeNormalVectors( const vec3_t forward, vec3_t right, vec3_t up );
 //int	PlaneTypeForNormal (vec3_t normal);
 
 void MatrixMultiply(float in1[3][3], float in2[3][3], float out[3][3]);
-void AngleVectors( const vec3_t angles, vec3_t forward, vec3_t right, vec3_t up);
+void Math_AnglesToVectors( const vec3_t angles, vec3_t forward, vec3_t right, vec3_t up);
```
Comment (D4edalus) [inline plugins/legacybanlist/q_shared.h]: would be better to keep it on AngleVectors to be able to reference it in q3 sources

**Comment by D4edalus** [comment]
pushed to a new branch for now, good work

**Pull request closed without merging**